src/Datadog.Trace.ClrProfiler.Managed/ScopeDBFactory.cs

```diff
@@ -0,0 +1,80 @@
+using System;
+using System.Data;
+using Datadog.Trace.ClrProfiler.Integrations.AdoNet;
+using Datadog.Trace.ExtensionMethods;
+using Datadog.Trace.Logging;
+
+namespace Datadog.Trace.ClrProfiler
+{
+    internal static class ScopeDBFactory<T>
+    {
+        private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor(typeof(ScopeDBFactory<T>));
+
+        private static readonly string _fullName;
+        private static readonly string _dbTypeName;
+        private static readonly string _operationName;
+
+        static ScopeDBFactory()
+        {
+            var type = typeof(T);
+            _fullName = type.FullName;
+            _dbTypeName = ScopeFactory.GetDbType(type.Name);
+            _operationName = $"{_dbTypeName}.query";
+        }
+
+        public static Scope CreateDbCommandScope(Tracer tracer, IDbCommand command)
+        {
+            if (_dbTypeName == null)
+            {
+                // don't create a scope, skip this span
+                return null;
+            }
+
+            if (!tracer.Settings.IsIntegrationEnabled(AdoNetConstants.IntegrationId))
+            {
+                // integration disabled, don't create a scope, skip this trace
+                return null;
+            }
+
+            if (tracer.Settings.AdoNetExcludedTypes.Count > 0 && tracer.Settings.AdoNetExcludedTypes.Contains(_fullName))
+            {
+                // AdoNet type disabled, don't create a scope, skip this trace
+                return null;
+            }
+
+            Scope scope = null;
+
+            try
+            {
+                Span parent = tracer.ActiveScope?.Span;
```

Comment:
I know we haven't done this before, but since users can set `Tracer.Instance = null` using the nuget, we should probably be more defensive about `tracer == null`. What do you think?

Reply:
Could be a separate PR where we do something in ALL integrations. Probably disable auto-instrumentation completely if `Tracer.Instance == null`.

Reply:
yeah, I agree we should have that check. I'll prepare a separated PR for that.